rules-reviewed/quarkus1/quarkus1-10/quarkus1-11-kubernetes-client.windup.groovy

```diff
@@ -0,0 +1,56 @@
+import org.jboss.windup.config.GraphRewrite
+import org.jboss.windup.config.metadata.TechnologyReference
+import org.jboss.windup.config.operation.iteration.AbstractIterationOperation
+import org.jboss.windup.config.query.Query
+import org.jboss.windup.config.query.QueryPropertyComparisonType
+import org.jboss.windup.graph.model.FileLocationModel
+import org.jboss.windup.graph.model.WindupConfigurationModel
+import org.jboss.windup.graph.model.resource.FileModel
+import org.jboss.windup.graph.service.GraphService
+import org.jboss.windup.graph.service.WindupConfigurationService
+import org.jboss.windup.reporting.category.IssueCategory
+import org.jboss.windup.reporting.category.IssueCategoryRegistry
+import org.jboss.windup.reporting.config.Hint
+import org.jboss.windup.reporting.config.Link
+import org.jboss.windup.rules.apps.java.condition.SourceMode
+import org.ocpsoft.rewrite.context.EvaluationContext
+
+final IssueCategory potentialIssueCategory = new IssueCategoryRegistry().getByID(IssueCategoryRegistry.POTENTIAL)
+
+ruleSet("quarkus1-11-kubernetes-client-groovy")
+        .addSourceTechnology(new TechnologyReference("quarkus1", "(,10]"))
+        .addTargetTechnology(new TechnologyReference("quarkus1", "[11,)")) 
+        .addRule()
+        .when(SourceMode.isDisabled(),
+                Query.fromType(FileModel)
+                .withProperty(FileModel.IS_DIRECTORY, Boolean.TRUE)
+                .withProperty(FileModel.FILE_PATH, QueryPropertyComparisonType.REGEX, ".*/io/quarkus/kubernetes/client\$"))
+        .perform(new AbstractIterationOperation<FileModel>() {
+            void perform(GraphRewrite event, EvaluationContext context, FileModel payload) {
+                //ensure that the dependency being searched for is present in the current application
+                //rather than any application in the MTA project
+                final String sourceBasePath = payload.getFilePath().replace("/io/quarkus/kubernetes/client", "")
+                final String dependencyJarName = sourceBasePath.substring(sourceBasePath.lastIndexOf("/") + 1)
+                WindupConfigurationModel windupConfigurationModel = WindupConfigurationService.getConfigurationModel(event.getGraphContext())
+                boolean packageComesFromAnalyzedApplication = false
+                windupConfigurationModel.getInputPaths().each {
+                    if (!packageComesFromAnalyzedApplication && it.filePath.endsWith(dependencyJarName)) packageComesFromAnalyzedApplication = true
+                }
+                if (!packageComesFromAnalyzedApplication) return
```

Comment:
Is this code simply trying to find if **any** element of windupConfigurationModel.getInputPaths() filePath.endsWith(dependencyJarName) ?
should it be easier to express it like this ?: 
```suggestion
                if (windupConfigurationModel.getInputPaths().any(it.filePath.endsWith(dependencyJarName)) return
```

Reply:
Hi Jonathan, Thanks for your feedback and suggestion. This logic has been copied from the quarkus rules generated. For example https://github.com/windup/windup-rulesets/blob/master/rules-generated/quarkus/avro/avro-to-quarkus.windup.groovy#L35-#L39 I'm confident the logic works in identifying the folder path we are looking for and ensuring it is present in the application being anaylzed, not just any application in the project. So unless there is a bug, and it doesn't work, then I'd like to leave it alone and keep that consistency with the generated code.